sql-plugin/src/main/scala-2.13/com/nvidia/spark/rapids/GpuBaseAggregateHelper.scala

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids
+
+import org.apache.spark.sql.catalyst.expressions.aggregate._
+
+object GpuBaseAggregateHelper {
+  def getAggPatternsCanReplace(strPatternToReplace: String): Array[Set[AggregateMode]] = {
```

Comment:
why are there two `gpuAggPatternsCanReplace`? It seems the only difference is the return type? Can we just add it to the regular one and remove these special cases?

Reply:
See comment. It's not possible to remove this special case because of the nature of the different Scala compilers

Reply:
Fixed